src/schoology-extractor/schoology_extractor/mapping/attendance.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pandas as pd
+
+from . import constants
+
+
+def _flatten_into_dataframe(attendance: list) -> pd.DataFrame:
+    df = pd.DataFrame(columns=["enrollment_id", "EventDate", "AttendanceStatus"])
+
+    for date_node in attendance:
+        if "statuses" not in date_node or "status" not in date_node["statuses"]:
+            continue
+
+        for item in date_node["statuses"]["status"]:
+            if "attendances" not in item or "attendance" not in item["attendances"]:
+                continue
+
+            reshape = pd.DataFrame(
+                [
+                    {
+                        "enrollment_id": a["enrollment_id"],
+                        "EventDate": date_node["date"],
+                        "AttendanceStatus": a["status"],
+                    }
+                    for a in item["attendances"]["attendance"]
+                ]
+            )
+
+            df = df.append(reshape)
+
+    return df.convert_dtypes()
+
+
+def __get_status(status_code: int) -> str:
+    switcher = {1: "present", 2: "absent", 3: "late", 4: "excused"}
+    return switcher.get(status_code, f"Unknown status: {status_code}")

```

Comment:
I have used underscore inconsistently here: just a single underscore for the first method, and double underscore for the second. Why did it happen? Indecisiveness. In both cases I have no need to directly unit test the function. They are both small local/private helper functions that keep `map_to_udm` from becoming too large. @lalrg @bradbanister do you have a preference between single and double underscore for functions like these? I've read about the difference before. At this point it feels more like a style decision to me, unless for some reason the "private" function is going to be mocked - then single underscore might be preferable?

(Why would you mock a helper function? Because that helper function is "hiding" access to something much more difficult to mock, such as a small routine for database access).

Reply:
Funny, I was thinking about this situation yesterday.  I'm fine with single underscore in your double underscore case.  I'm a fan of method extraction, often just to give a name to a particular operation being performed.  I think it's ok not to try to differentiate the reason a function is private.

Reply:
I like single underscore.

Reply:
Going to change to single underscore.

Reply:
Done